remove hardcoded dimensions from all pages

oshorefueled opened the pull request:
#### Resolves

Issue #161 

#### What's new

This PR adds uniform padding across all pages. It also add removes hard-coded margin and padding dimensions, adding standard dimensions and text sizes.

itswisdomagain commented on nuklear/desktop.go:
```diff
@@ -8,45 +8,48 @@ import (
+	handlers := getHandlers()
+	desktop.handlers = make(map[string]Handler, len(handlers))
+	for _, handler := range handlers {
+		handler.handler.SetWalletMiddleware(walletMiddleware)
```
No need to maintain an instance of `walletMiddleware` in all handlers. Remove the `handler.SetWalletMiddleware ` function and modify the `handler.Render` function to accept `wallet walletcore.Wallet`. Whenever you call `handler.Render`, pass `desktop.wallet` as parameter to the function.

itswisdomagain commented on nuklear/desktop.go:
```diff
@@ -57,87 +60,65 @@ func LaunchApp(ctx context.Context, walletMiddleware app.WalletMiddleware) error
-	w.LayoutSpacePushScaled(rect)
+	w.LayoutSpacePushScaled(contentRect)
+	handler, ok := desktop.handlers[desktop.currentPage]
+	if !ok { // ideally, this should only be false once in the lifetime of an instance
```
instead of this, it would be better to check if `desktop.currentPage == ""` which is the only reason why `ok == false`
If `desktop.currentPage == ""`, change page to home page

itswisdomagain commented on nuklear/desktop.go:
```diff
@@ -57,87 +60,65 @@ func LaunchApp(ctx context.Context, walletMiddleware app.WalletMiddleware) error
+func (desktop *Desktop) changePage(page string) {
+	desktop.currentPage = page
+	desktop.pageChanged = true
+	desktop.masterWindow.Changed()
```
I have some concern about having to draw the navigation section everytime the page changes. Might be better to add the content window to desktop (`desktop.pageWindow`) and update that instead

itswisdomagain commented on nuklear/handler.go:
```diff
@@ -0,0 +1,44 @@
+type Handler interface {
+	BeforeRender()
+	Render(*nucular.Window)
+	SetWalletMiddleware(walletcore.Wallet)
```
As explained above, this should go and `Render` should be modified to accept `walletcore.Wallet`

itswisdomagain commented on nuklear/handlers/transactions.go:
```diff
@@ -0,0 +1,75 @@
+	hasFetchedTransactions bool
+}
+
+func (handler *TransactionsHandler) fetchTransactions() {
```
This method is called by `Render`, it should appear below the `Render` function

itswisdomagain commented on nuklear/desktop.go:
```diff
@@ -57,87 +60,65 @@ func LaunchApp(ctx context.Context, walletMiddleware app.WalletMiddleware) error
+	w.LayoutSpacePushScaled(navRect)
+	// render nav
+	helpers.SetNavStyle(desktop.masterWindow)
 	if sw := w.GroupBegin("Navigation Group", 0); sw != nil {
```
- You can use `helpers.NewWindow` here
- What does `sw` mean? Maybe use a more descriptive variable name
- If `sw` is nil, then we have no navigation window, without a navigation window, the app is not usable, this scenario should be attended to, perhaps show an error message

itswisdomagain commented on nuklear/helpers/style.go:
```diff
@@ -55,63 +66,42 @@ var colorTable = nstyle.ColorTable{
+	fontInit.Do(func() {
+		fontData, err := ioutil.ReadFile(asset)
+		if err != nil {
+			log.Fatal(err)
```
Interfaces should ideally propagate errors back to main.go so that the loaded wallet is shutdown properly. The following modifications should make this work.

```
func InitFonts() (err error) {
	robotoMediumFont, err := getFont(navFontSize, navFontDPI, "nuklear/assets/font/Roboto-Medium.ttf")
	if err != nil {
		return
	}
	PageHeaderFont, NavFont = robotoMediumFont, robotoMediumFont

	PageContentFont, err = getFont(pageContentFontSize, pageContentFontDPI, "nuklear/assets/font/Roboto-Light.ttf")
	return
}

func getFont(fontSize, DPI int, asset string) (font.Face, error) {
	fontData, err := ioutil.ReadFile(asset)
	if err != nil {
		return nil, err
	}

	ttfont, err := freetype.ParseFont(fontData)
	if err != nil {
		return nil, err
	}

	size := int(float64(fontSize) * scaling)
	options := &truetype.Options{
		Size:    float64(size),
		Hinting: font.HintingFull,
		DPI:     float64(DPI),
	}

	return truetype.NewFace(ttfont, options), nil
}
```
#### Key changes
- Modify `InitFonts` and `getFont` to return `error`
- Modify line 42 of `nuklear/desktop.go` to propagate any `InitFonts` error
```
err := helpers.InitFonts()
if err != nil {
	return err
}
```

#### Other things to note:
- Don't see a need for `sync.Once`, the function is only called once when the app is started
- `PageHeaderFont` and `NavFont` use the same roboto-medium font, no need to re-read the font file multiple times
- `InitFonts` calls `getFont`, the functions should be arranged in that order for easier scanning

itswisdomagain commented on nuklear/helpers/style.go:
```diff
@@ -66,18 +64,44 @@ var colorTable = nstyle.ColorTable{
+	return nil
+}
+
+func readFontFile(file string) ([]byte, error) {
```
can just use `ioutil.ReadFile` directly instead of creating a wrapper around the function

itswisdomagain commented on nuklear/helpers/style.go:
```diff
@@ -66,18 +64,44 @@ var colorTable = nstyle.ColorTable{
+		return err
+	}
+
+	NavFont, err = getFont(navFontSize, navFontDPI, robotoMediumFontData)
```
I prefer the previous `getFont` function that takes the asset string rather than this, above, there are 2 references to `readFontFile` which really is `ioutil.ReadFile` - a repetition that can be avoided if the call to `ioutil.ReadFile` is made in the `getFont` function

oshorefueled committed:
add uniform padding to all pages

* add dimensions.go to centralize dimensions
* remove individual margins from pages

oshorefueled committed:
remove hard-coded dimensions and text sizes

clean padding and margin on wallets page

clean padding dimensions on receive page

clean up padding dimensions on transactions page

clean up padding dimensions on transaction details

cleanup padding dimension on create page

clean up dimensions

* clean up dimensions for sign message page
* clean up dimension for verify message page

clean dimension padding for sub pages

* clean up dimensions for accounts page
* clean up dimensions for passphrase page

git import code

clean up text sizes

move unique dimensions to dimensions.go

clean up padding dimensions for backup page

remove comments from dimensions

remove hard coded dimension in send page

oshorefueled committed:
trigger github actions
